File: src/main/java/ghx/mandelbrot/DoBindings.java
package ghx.mandelbrot;

import javafx.beans.binding.Bindings;
import javafx.beans.property.DoubleProperty;
import javafx.beans.property.StringProperty;
import javafx.util.StringConverter;

public class DoBindings {
    static void bind(StringProperty s, DoubleProperty v) {
        Bindings.bindBidirectional(
                s,
                v,
                new StringConverter<Number>() {

                    public Double fromString(String s) {
//                        System.out.println("from string " + s);
                        try {
                            return Double.parseDouble(s);
                        } catch (Throwable t) {
                            return 0.0;
                        }
                    }

                    public String toString(Number n) {
//                        System.out.println("from number " + n);
                        if (((Double) n).isInfinite())
                            return "Infinity";
                        else
                            return n.toString();
                    }
                }
                );
    }
}

File: src/main/scala/ghx/mandelbrot/DoBindings.scala
package ghx.mandelbrot

import javafx.beans.binding.Bindings
import javafx.beans.property.{DoubleProperty, StringProperty}
import javafx.util.StringConverter

object DoBindings {
  private[mandelbrot] def bind(s: StringProperty, v: DoubleProperty): Unit = {
    Bindings.bindBidirectional(s, v, new StringConverter[Number]() {
      override def fromString(s: String): Number = { //                        System.out.println("from string " + s);
        try  s.toDouble
        catch {
          case t: NumberFormatException => 0.0
        }
      }

      override def toString(n: Number): String = { //                        System.out.println("from number " + n);
        n match {
          case d: java.lang.Double if d.isInfinite => "Infinity"
          case x => x.toString
        }
      }
    })
  }
}
